src/sidebar/components/FilterStatus.js

```diff
@@ -49,271 +26,234 @@ import { useRootThread } from './hooks/use-root-thread';
-        {store.isLoading() ? (
-          <Spinner />
-        ) : (
-          <>
-            <div
-              className={classNames(
-                // Setting `min-width: 0` here allows wrapping to work as
-                // expected for long `filterQuery` strings. See
-                // https://css-tricks.com/flexbox-truncated-text/
-                'grow min-w-[0]'
-              )}
-              data-testid="filter-text"
-            >
-              {resultCount > 0 && <span>Showing </span>}
-              <span className="whitespace-nowrap font-bold">
-                {resultCount > 0 ? resultCount : 'No'}{' '}
-                {resultCount === 1 ? entitySingular : entityPlural}
-              </span>
-              {filterQuery && (
-                <span>
-                  {' '}
-                  for <span className="break-words">{`'${filterQuery}'`}</span>
-                </span>
-              )}
-              {focusDisplayName && (
-                <span>
-                  {' '}
-                  by{' '}
-                  <span className="whitespace-nowrap font-bold">
-                    {focusDisplayName}
-                  </span>
-                </span>
-              )}
-              {additionalCount > 0 && (
-                <span className="whitespace-nowrap italic text-color-text-light">
-                  {' '}
-                  (and {additionalCount} more)
-                </span>
-              )}
-            </div>
-            <div>{actionButton}</div>
-          </>
-        )}
-      </div>
-    </Card>
+    <>
+      {resultCount > 0 && <span>Showing </span>}
+      <span className="whitespace-nowrap font-bold">
+        {resultCount > 0 ? resultCount : 'No'}{' '}
+        {resultCount === 1 ? entitySingular : entityPlural}
+      </span>
+      {filterQuery && (
+        <span>
+          {' '}
+          for <span className="break-words">{`'${filterQuery}'`}</span>
+        </span>
+      )}
+      {focusDisplayName && (
+        <span>
+          {' '}
+          by{' '}
+          <span className="whitespace-nowrap font-bold">
+            {focusDisplayName}
+          </span>
+        </span>
+      )}
+      {additionalCount > 0 && (
+        <span className="whitespace-nowrap italic text-color-text-light">
+          {' '}
+          (and {additionalCount} more)
+        </span>
+      )}
+    </>
   );
 }
 
 /**
- * This status is used when there are selected annotations (including direct-
- * linked annotations). This status takes precedence over others.
+ * Show a description of currently-applied filters and a button to clear the
+ * filter(s).
  *
- * Message formatting:
- * "[Showing] (No|<resultCount>) annotation[s] [\(and <additionalCount> more\)]"
- * Button:
- * "<cancel icon> Show all [\(<totalCount)\)]" - clears the selection
+ * There are four filter modes. Exactly one is applicable at any time. In order
+ * of precedence:
```

Comment:
A high level summary of the four modes and their precedence is useful, but I suspect the details are liable to get out of sync with the implementation.

Reply:
Yeah. I had to choose between keeping the chunks and bits documented near their use (accurate but utterly fails to give the full picture) or this higher-level composite (quite a bit more helpful but easy to get out-of-sync). I chose the latter...